services/fetchDictionaryData.js

```diff
@@ -0,0 +1,35 @@
+const DICTIONARY_API_URL = "https://api.dictionaryapi.dev/api/v2/entries/en/";
+
+export async function getPronunciationAudio(word) {
+  const wordForPath = word.trim().toLowerCase();
+  try {
+    const response = await fetch(DICTIONARY_API_URL + wordForPath);
+    if (response.ok) {
+      const data = await response.json();
+      const urls = extractAudioLinks(data);
+      return addCountryCodesToLinks(urls);
+    }
+  } catch (error) {
+    console.error(error.message);
+    return [];
+  }
+}
+
+function extractAudioLinks(data) {
```

Comment:
never heard of ``flatMap``. Sounds good and I am looking forward to hear about that. 👍